base/initdefs.jl

```diff
@@ -18,10 +18,27 @@ An array of the command line arguments passed to Julia, as strings.
 """
 const ARGS = String[]
 
+"""
+    exit([code])
+
+Quit (or control-D at the prompt). The default exit code is zero, indicating that the
+processes completed successfully.
+"""
 exit(n) = ccall(:jl_exit, Void, (Int32,), n)
 exit() = exit(0)
+
+"""
+    quit()
+
+Quit the program indicating that the processes completed successfully. This function calls
+`exit(0)` (see [`exit`](@ref)).
+"""
 quit() = exit()
 
+# hint for people who forget parentheses
```

Comment:
I'm actually talking about something more integrated into the REPL (since that's where we need this) instead of modifying how some functions are displayed.

Something like

```diff
diff --git a/base/REPL.jl b/base/REPL.jl
index 7a4eb61b46..5ac8261dfc 100644
--- a/base/REPL.jl
+++ b/base/REPL.jl
@@ -138,7 +138,9 @@ function print_response(errio::IO, val::ANY, bt, show_value::Bool, have_color::B
                                                           errio, QuoteNode(val), bt))))
                 iserr, lasterr = false, ()
             else
-                if val !== nothing && show_value
+                if (val === exit || val === quit) && show_value && specialdisplay === nothing
+                    println("Use exit(), quit() or Ctrl-D to exit")
+                elseif val !== nothing && show_value
                     try
                         if specialdisplay === nothing
                             eval(Main, Expr(:body, Expr(:return, Expr(:call, display, QuoteNode(val)))))
```

This gives;
```julia
julia> exit
Use exit(), quit() or Ctrl-D to exit

julia> quit
Use exit(), quit() or Ctrl-D to exit

julia> print(exit)
exit
julia> display(exit)
exit (generic function with 2 methods)

julia> display(quit)
quit (generic function with 1 method)
```

Ideally we may want to do this only when the input is just `quit` and `exit`.

When trying this I realized that this will make it harder to see how many methods are defined for the two function. Hopefully it's fine...